RFC/src/RFC-0120_Consensus.md

```diff
@@ -0,0 +1,212 @@
+
+THIS DOCUMENT IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS", AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", 
+"NOT RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in 
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
+shown here.
+
+## Disclaimer
+
+This document and its content are intended for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community of the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+The aim of this Request for Comment (RFC) is to describe the fields that a block should contain as well as all consensus rules that will determine the validity of a block.
+
+## Related Requests for Comment
+
+* [RFC-0100: Base Layer](RFC-0100_BaseLayer.md)
+* [RFC-0130: Mining](./RFC-0130_Mining.md)
+* [RFC-0140: SyncAndSeeding](RFC-0140_Syncing_and_seeding.md)
+
+## Description
+
+### Blocks
+
+Every [block] MUST conform to the following:
+
+* Have a _single_ valid coinbase [UTXO] and kernel
+* have a _single_ valid blockheader;
+* [cut-through] has been applied where possible;
+* every [UTXO] has a valid [range proof].
+
+If a [block] does not confirm to the above, the block should be rejected as invalid and the peer from which it was received marked as a malicious peer.
+
+### Blockheaders
+
+Every [block header] MUST contain the following fields:
+
+* version;
+* height;
+* prev_hash;
+* timestamp;
+* output_mr;
+* range_proof_mr;
+* kernel_mr;
+* total_kernel_offset;
+* total_difficulty;
+* nonce;
+* pow.
+
+The [block header] MUST confirm to the following:
+
+* The nonce and [PoW](#pow) must be a valid for the [block header].
+* All the merkle roots must be valid for the states after the [block] was applied to the local state.
+
+If the [block header] does not confirm to any of the above, the [block] needs to be rejected and the peer the it was received from, marked as a malicious peer.
```

Comment:
```suggestion
If the [block header] does not confirm to any of the above, the [block] MUST be rejected.
If a peer rejects multiple blocks from a given peer, it MAY blacklist the peer and ignore further communication from it.
```